servicetalk-http-api/src/main/java/io/servicetalk/http/api/SingleAddressHttpClientBuilder.java

```diff
@@ -144,4 +144,29 @@
      * @return {@code this}.
      */
     public abstract SingleAddressHttpClientBuilder<U, R> sslConfig(ClientSslConfig sslConfig);
+
+    /**
+     * Toggle inference of value to use instead of {@link ClientSslConfig#peerHost()}
+     * from client's address when peer host is not specified. By default, inference is enabled.
+     * @param shouldInfer value indicating whether inference is on ({@code true}) or off ({@code false}).
+     * @return {@code this}
+     */
+    public abstract SingleAddressHttpClientBuilder<U, R> inferPeerHost(boolean shouldInfer);
```

Comment:
We previously discussed in https://github.com/apple/servicetalk/pull/1561#discussion_r632269087, but doesn't look like we come to a conclusion. Does it make sense to join peer host and port together on `ClientSslConfig` API into a single method `HostAndPort peerHostAndPort()`? If yes, we can have a single method here: `inferPeerHostAndPort(boolean)` instead of two.

Reply:
@Scottmitch wdyt about single `ClientSslConfig#peerHostAndPort()` and inferPeerHostAndPort(boolean)?

Reply:
To my understanding the peer port is not used when peer host is not inferred. I think it's reasonable to provide an API where both are affected. And yes, the breaking API change would need to go to `ClientSslConfig`. Perhaps we could do it in a separate PR and for now add this feature with no breakage?

Reply:
I was thinking that if we merge host and port into one method, we can provide only one "infer" method and won't need to change client API later.
Let's make the final decision tomorrow morning. Everything else looks good in the PR.

Reply:
Discussed with @Scottmitch offline. He suggested to keep host and port independent because sometimes users don't know what is the port number (SRV resolutions or use of Unix Domain Sockets).